cdap-master/src/test/java/co/cask/cdap/master/environment/k8s/MasterServiceMainTestBase.java

```diff
@@ -71,34 +72,35 @@
 
   private static InMemoryZKServer zkServer;
   private static Map<Class<?>, ServiceMainManager<?>> serviceManagers = new LinkedHashMap<>();
+  private static CConfiguration originalCConf;
 
   @BeforeClass
   public static void init() throws Exception {
     zkServer = InMemoryZKServer.builder().setAutoCleanDataDir(false).setDataDir(TEMP_FOLDER.newFolder()).build();
     zkServer.startAndWait();
 
-    CConfiguration cConf = CConfiguration.create();
+    originalCConf = CConfiguration.create();
 
     // Set the HDFS directory as well as we are using DFSLocationModule in the master services
-    cConf.set(Constants.CFG_HDFS_NAMESPACE, TEMP_FOLDER.newFolder().getAbsolutePath());
-    cConf.set(Constants.Zookeeper.QUORUM, zkServer.getConnectionStr());
+    originalCConf.set(Constants.CFG_HDFS_NAMESPACE, TEMP_FOLDER.newFolder().getAbsolutePath());
+    originalCConf.set(Constants.Zookeeper.QUORUM, zkServer.getConnectionStr());
 
     // Set all bind address to localhost
     String localhost = InetAddress.getLoopbackAddress().getHostName();
     StreamSupport.stream(CConfiguration.create().spliterator(), false)
       .map(Map.Entry::getKey)
       .filter(s -> s.endsWith(".bind.address"))
-      .forEach(key -> cConf.set(key, localhost));
+      .forEach(key -> originalCConf.set(key, localhost));
 
     // Set router to bind to random port
-    cConf.setInt(Constants.Router.ROUTER_PORT, 0);
+    originalCConf.setInt(Constants.Router.ROUTER_PORT, 0);
 
     // Start the master main services
-    serviceManagers.put(RouterServiceMain.class, runMain(RouterServiceMain.class, cConf));
-    serviceManagers.put(MessagingServiceMain.class, runMain(MessagingServiceMain.class, cConf));
-    serviceManagers.put(MetricsServiceMain.class, runMain(MetricsServiceMain.class, cConf));
-    serviceManagers.put(MetadataServiceMain.class, runMain(MetadataServiceMain.class, cConf));
-    serviceManagers.put(AppFabricServiceMain.class, runMain(AppFabricServiceMain.class, cConf));
+    serviceManagers.put(RouterServiceMain.class, runMain(RouterServiceMain.class, null));
```

Comment:
What's the point of having all of them `null`?

Reply:
fixed